diff --git a/Cargo.toml b/Cargo.toml
index b2865f9af30d792109e9b2d91bfe8c803c15ba79..bc6e5aff18d50475eaa55ae64597da8f11a69efa 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,6 +23,7 @@ dbg_macro = "deny"
 todo = "deny"
 
 [workspace.dependencies]
+stacker = "0.1.25"
 thiserror = { version = "2.0.18" }
 tokio = { version = "1.52.3", features = ["full"] }
 tokio-util = { version = "0.7.19", features = ["rt"] }
@@ -162,7 +163,7 @@ prost-types = "0.14"
 # The `axum` version must match the one used in `tonic` (replace `RELEASE` with the release we are using):
 #   https://github.com/hyperium/tonic/blob/vRELEASE/tonic/Cargo.toml
 axum = "0.8.9"
-datafusion = { version = "55.1.0", default-features = false, features = [
+datafusion = { version = "=55.1.0", default-features = false, features = [
     "nested_expressions",
     "crypto_expressions",
     "datetime_expressions",
@@ -176,32 +177,33 @@ datafusion = { version = "55.1.0", default-features = false, features = [
     "avro",
     "recursive_protection",
 ] }
-datafusion-common = { version = "55.1.0", features = ["object_store"] }
-datafusion-datasource = { version = "55.1.0" }
-datafusion-datasource-avro = { version = "55.1.0" }
-datafusion-datasource-json = { version = "55.1.0" }
-datafusion-datasource-parquet = { version = "55.1.0" }
-datafusion-expr = { version = "55.1.0", default-features = false }
-datafusion-expr-common = { version = "55.1.0" }
-datafusion-proto = { version = "55.1.0" }
-datafusion-functions = { version = "55.1.0" }
-datafusion-functions-nested = { version = "55.1.0", default-features = false }
-datafusion-physical-expr = { version = "55.1.0" }
-datafusion-session = { version = "55.1.0" }
-datafusion-spark = { version = "55.1.0", features = ["core"] }
+datafusion-common = { version = "=55.1.0", features = ["object_store"] }
+datafusion-datasource = { version = "=55.1.0" }
+datafusion-datasource-avro = { version = "=55.1.0" }
+datafusion-datasource-json = { version = "=55.1.0" }
+datafusion-datasource-parquet = { version = "=55.1.0" }
+datafusion-expr = { version = "=55.1.0", default-features = false }
+datafusion-expr-common = { version = "=55.1.0" }
+datafusion-ffi = { version = "=55.1.0" }
+datafusion-proto = { version = "=55.1.0" }
+datafusion-functions = { version = "=55.1.0" }
+datafusion-functions-nested = { version = "=55.1.0", default-features = false }
+datafusion-physical-expr = { version = "=55.1.0" }
+datafusion-session = { version = "=55.1.0" }
+datafusion-spark = { version = "=55.1.0", features = ["core"] }
 pyo3 = { version = "0.29.0", features = ["serde"] }
 jiter = { version = "0.15.0", default-features = false }
-arrow = { version = "59.2.0", features = ["chrono-tz"] }
-arrow-array = { version = "59.2.0", features = ["ffi"] }
-arrow-buffer = { version = "59.2.0" }
-arrow-data = { version = "59.2.0" }
-arrow-schema = { version = "59.2.0", features = ["serde"] }
-arrow-flight = { version = "59.2.0", features = ["flight-sql-experimental"] }
-arrow-pyarrow = { version = "59.2.0" }
-parquet = { version = "59.2.0" }
-parquet-variant = { version = "59.2.0" }
-parquet-variant-compute = { version = "59.2.0" }
-parquet-variant-json = { version = "59.2.0" }
+arrow = { version = "=59.3.0", features = ["chrono-tz"] }
+arrow-array = { version = "=59.3.0", features = ["ffi"] }
+arrow-buffer = { version = "=59.3.0" }
+arrow-data = { version = "=59.3.0" }
+arrow-schema = { version = "=59.3.0", features = ["serde"] }
+arrow-flight = { version = "=59.3.0", features = ["flight-sql-experimental"] }
+arrow-pyarrow = { version = "=59.3.0" }
+parquet = { version = "=59.3.0" }
+parquet-variant = { version = "=59.3.0" }
+parquet-variant-compute = { version = "=59.3.0" }
+parquet-variant-json = { version = "=59.3.0" }
 serde_arrow = { version = "0.14.2", features = ["arrow-59"] }
 # The `object_store` version must match the one used in DataFusion.
 object_store = { version = "0.13.2", features = ["aws", "gcp", "azure", "http"] }
