diff --git a/crates/sail-common-datafusion/Cargo.toml b/crates/sail-common-datafusion/Cargo.toml
index 3e5373a5690077a161d2221d8b81fddca9618ed1..85c1fd71a06c7d21ab228c91920591f2c7349f5e 100644
--- a/crates/sail-common-datafusion/Cargo.toml
+++ b/crates/sail-common-datafusion/Cargo.toml
@@ -8,6 +8,7 @@ workspace = true
 
 [dependencies]
 sail-common = { path = "../sail-common" }
+sail-native-resource-ffi = { path = "../sail-native-resource-ffi" }
 
 chrono = { workspace = true }
 datafusion = { workspace = true }
