diff --git a/crates/sail-common-datafusion/src/lib.rs b/crates/sail-common-datafusion/src/lib.rs
index e8b8fbf2c8bf7f5becbc6e2f04cdb6f84405982e..eb962e018634ee9bca36950d1300f65b0fbeb4aa 100644
--- a/crates/sail-common-datafusion/src/lib.rs
+++ b/crates/sail-common-datafusion/src/lib.rs
@@ -1,11 +1,13 @@
 pub mod array;
 pub mod catalog;
 pub mod column_features;
+pub mod connect_extension;
 pub mod datasource;
 pub mod display;
 pub mod error;
 pub mod extension;
 pub mod formatter;
+pub mod geometry;
 mod java_float;
 pub mod lakesource;
 pub mod literal;
@@ -20,3 +22,7 @@ pub mod streaming;
 pub mod udf;
 pub mod utils;
 pub mod variant;
+
+pub mod driver_extension;
+pub mod native_resource;
+pub mod native_scalar;
