diff --git a/crates/sail-common-datafusion/src/session/mod.rs b/crates/sail-common-datafusion/src/session/mod.rs
index 2bde515e56dcdca5ef2ed4af8e0464c8799e8a39..23e61281b8659a968e1632901e8a044c1450e81c 100644
--- a/crates/sail-common-datafusion/src/session/mod.rs
+++ b/crates/sail-common-datafusion/src/session/mod.rs
@@ -1,4 +1,5 @@
 pub mod activity;
 pub mod job;
+pub mod lifecycle;
 pub mod plan;
 pub mod repartition;
