diff --git a/crates/sail-execution/src/proto/mod.rs b/crates/sail-execution/src/proto/mod.rs
index 49599ef748d665216da9efc52a964c861bbadb14..e1448edc33157b2c4872d445881fafaa7fe23075 100644
--- a/crates/sail-execution/src/proto/mod.rs
+++ b/crates/sail-execution/src/proto/mod.rs
@@ -2,6 +2,7 @@ mod codec;
 mod converter;
 mod decode;
 mod encode;
+mod native_expr;
 
 pub use codec::RemoteExecutionCodec;
 #[cfg(test)]
