diff --git a/crates/sail-native-resource-ffi/Cargo.toml b/crates/sail-native-resource-ffi/Cargo.toml
new file mode 100644
index 0000000000000000000000000000000000000000..b09f272ab275bd1cbd8b00a90303a4ad9517a39d
--- /dev/null
+++ b/crates/sail-native-resource-ffi/Cargo.toml
@@ -0,0 +1,9 @@
+[package]
+name = "sail-native-resource-ffi"
+version = "0.1.0"
+edition = "2024"
+license = "Apache-2.0"
+publish = false
+
+[lints]
+workspace = true
