diff --git a/crates/sail-session/Cargo.toml b/crates/sail-session/Cargo.toml
index c959dda31102b60919cddee8ac9b7b58021e8d66..581934556b4aa70bfa0a3e121d8b1cc50214a279 100644
--- a/crates/sail-session/Cargo.toml
+++ b/crates/sail-session/Cargo.toml
@@ -41,6 +41,11 @@ datafusion = { workspace = true }
 datafusion-common = { workspace = true }
 datafusion-datasource = { workspace = true }
 datafusion-expr = { workspace = true }
+datafusion-ffi = { workspace = true }
+arrow-schema = { workspace = true }
+pyo3 = { workspace = true }
+serde = { workspace = true }
+serde_json = { workspace = true }
 datafusion-physical-expr = { workspace = true }
 object_store = { workspace = true }
 secrecy = { workspace = true }
@@ -50,6 +55,11 @@ chrono = { workspace = true }
 indexmap = { workspace = true }
 futures = { workspace = true }
 uuid = { workspace = true }
+prost = { workspace = true }
+url = { workspace = true }
+
+[build-dependencies]
+prost-build = { workspace = true }
 
 [dev-dependencies]
 tempfile = { workspace = true }
