diff --git a/crates/sail-session/src/session_manager/actor/mod.rs b/crates/sail-session/src/session_manager/actor/mod.rs
index 9061fc33ca14da714d6ce9e0497f69d04150488b..ec75defa80b6162a501a7533e241754d4fddac3c 100644
--- a/crates/sail-session/src/session_manager/actor/mod.rs
+++ b/crates/sail-session/src/session_manager/actor/mod.rs
@@ -11,6 +11,7 @@ use sail_execution::{DriverId, IdGenerator};
 use sail_telemetry::events::SystemEventReporter;
 
 use crate::session_factory::{ServerSessionInfo, SessionFactory, SessionJobRunnerFactory};
+use crate::session_manager::cleanup::SessionCleanup;
 use crate::session_manager::session::ServerSession;
 
 pub struct SessionManagerActor {
@@ -18,6 +19,7 @@ pub struct SessionManagerActor {
     session_factory: Box<dyn SessionFactory<ServerSessionInfo>>,
     job_runner_factory: Box<dyn SessionJobRunnerFactory>,
     sessions: IndexMap<String, ServerSession>,
+    cleanup: SessionCleanup,
     drivers: DriverRegistry,
     driver_gateway: Option<DriverGateway>,
     driver_id_generator: IdGenerator<DriverId>,
