diff --git a/crates/sail-spark-connect/Cargo.toml b/crates/sail-spark-connect/Cargo.toml
index 2b288f5ea39fbb38338379a6a1e055088a6eb5cc..0b79022cbb0a11582dfbaf3477728dcb600703f5 100644
--- a/crates/sail-spark-connect/Cargo.toml
+++ b/crates/sail-spark-connect/Cargo.toml
@@ -7,6 +7,7 @@ edition = { workspace = true }
 workspace = true
 
 [dependencies]
+stacker = { workspace = true }
 sail-cache = { path = "../sail-cache" }
 sail-common = { path = "../sail-common" }
 sail-common-datafusion = { path = "../sail-common-datafusion" }
