diff --git a/crates/sail-spark-connect/src/proto/mod.rs b/crates/sail-spark-connect/src/proto/mod.rs
index 1dc12e61045d0fb638b08ceddc9c9c05b1e57e65..2d5e6371388750f86218d641c2bd546505e4c5ab 100644
--- a/crates/sail-spark-connect/src/proto/mod.rs
+++ b/crates/sail-spark-connect/src/proto/mod.rs
@@ -4,6 +4,7 @@ pub(crate) mod data_type;
 pub(crate) mod data_type_arrow;
 pub(crate) mod data_type_json;
 pub(crate) mod expression;
+pub(crate) mod extension;
 pub(crate) mod function;
 pub(crate) mod literal;
 pub(crate) mod plan;
