diff --git a/crates/sail-spark-connect/src/service/plan_executor.rs b/crates/sail-spark-connect/src/service/plan_executor.rs
index 1c42e4eb2c46eaf2376003ff034ca0385c1657b2..d0e8d103079b01e8f32fe23721d5ca5391900559 100644
--- a/crates/sail-spark-connect/src/service/plan_executor.rs
+++ b/crates/sail-spark-connect/src/service/plan_executor.rs
@@ -566,9 +566,10 @@ pub(crate) async fn handle_execute_remove_cached_remote_relation_command(
 pub(crate) async fn handle_interrupt_all(ctx: &SessionContext) -> SparkResult<Vec<String>> {
     let spark = ctx.extension::<SparkSession>()?;
     let mut results = vec![];
-    for executor in spark.remove_all_executors()? {
-        executor.pause_if_running().await?;
-        results.push(executor.metadata.operation_id.clone());
+    for executor in spark.all_executors()? {
+        if executor.interrupt().await? {
+            results.push(executor.metadata.operation_id.clone());
+        }
     }
     Ok(results)
 }
@@ -579,9 +580,10 @@ pub(crate) async fn handle_interrupt_tag(
 ) -> SparkResult<Vec<String>> {
     let spark = ctx.extension::<SparkSession>()?;
     let mut results = vec![];
-    for executor in spark.remove_executors_by_tag(tag.as_str())? {
-        executor.pause_if_running().await?;
-        results.push(executor.metadata.operation_id.clone());
+    for executor in spark.executors_by_tag(tag.as_str())? {
+        if executor.interrupt().await? {
+            results.push(executor.metadata.operation_id.clone());
+        }
     }
     Ok(results)
 }
@@ -591,10 +593,13 @@ pub(crate) async fn handle_interrupt_operation_id(
     operation_id: String,
 ) -> SparkResult<Vec<String>> {
     let spark = ctx.extension::<SparkSession>()?;
-    match spark.remove_executor(operation_id.as_str())? {
+    match spark.get_executor(operation_id.as_str())? {
         Some(executor) => {
-            executor.pause_if_running().await?;
-            Ok(vec![executor.metadata.operation_id.clone()])
+            if executor.interrupt().await? {
+                Ok(vec![executor.metadata.operation_id.clone()])
+            } else {
+                Ok(vec![])
+            }
         }
         None => Ok(vec![]),
     }
